src/Storage/Storage.Management/StorageAccount/RestoreAzStorageBlobRange.cs

```diff
@@ -119,49 +121,80 @@ public override void ExecuteCmdlet()
             if (ShouldProcess(this.StorageAccountName, "Restore Blob Range"))
             {
                 if (waitForComplete)
-                {                    
-                    Task<AzureOperationResponse<BlobRestoreStatus>> beginTask = this.StorageClient.StorageAccounts.BeginRestoreBlobRangesWithHttpMessagesAsync(
-                     this.ResourceGroupName,
-                     this.StorageAccountName,
-                     this.TimeToRestore,
-                     PSBlobRestoreRange.ParseBlobRestoreRanges(this.BlobRestoreRange));
-
-                    beginTask.Wait();
-
-                    AzureOperationResponse<BlobRestoreStatus> response = beginTask.Result;
-
-                    WriteWarning(string.Format("Restore blob ranges with Id '{0}' started. Restore blob ranges time to complete is dependent on the size of the restore.", response.Body is null ? "" : response.Body.RestoreId));
+                {
+                    Track2.StorageAccountResource account = this.StorageClientTrack2.GetStorageAccount(this.ResourceGroupName, this.StorageAccountName);
+                    var restoreLro = account.RestoreBlobRanges(
+                        WaitUntil.Started,
+                        new Track2Models.BlobRestoreContent(
+                            this.TimeToRestore,
+                            ParseBlobRestoreRanges(this.BlobRestoreRange))
+                        );
+
+                    Dictionary<string, object> temp = (Dictionary<string, object>)restoreLro.GetRawResponse().Content.ToObjectFromJson();
+                    object restoreId;
+
+                    if (temp.TryGetValue("restoreId", out restoreId))
+                    {
+                        WriteWarning(string.Format("Restore blob ranges with Id '{0}' started. Restore blob ranges time to complete is dependent on the size of the restore.", restoreId));
+                    }
+                    else
+                    {
+                        WriteWarning(string.Format("Restore blob ranges with Id  started. Restore blob ranges time to complete is dependent on the size of the restore."));
```

Comment:
may be "without id", since `restoreId` is not fetched?

Reply:
Maybe we should remove "with Id" here. OR jut report error, since it should have the restore ID.

Reply:
Per discussion, removed "with id", and simply print a warning of "cannot fetch id" here.